File: apps/android/core/model/src/main/kotlin/com/scribblefit/core/model/Scribble.kt
package com.scribblefit.core.model

/**
 * Domain model representing a scribble.
 */
import kotlinx.serialization.Serializable

@Serializable
data class Scribble(
    val id: Long,
    val rawText: String,
    val status: ScribbleStatus,
    val createdAt: Long,
    val exercises: List<Exercise> = emptyList(),
)

enum class ScribbleStatus {
    PENDING,
    PARSING,
    SUCCESS,
    FAILED,
    COMPLETED,
}

File: apps/ios/ScribbleFit/Sources/Domain/Models/Scribble.swift
import Foundation

public struct Scribble: Identifiable, Equatable, Sendable {
    public let id: UUID
    public let rawText: String
    public let parsedJson: String?
    public let status: ScribbleStatus
    public let createdAt: Date
    public let exercises: [Exercise]

    public init(
        id: UUID = UUID(),
        rawText: String,
        parsedJson: String? = nil,
        status: ScribbleStatus = .raw,
        createdAt: Date = Date(),
        exercises: [Exercise] = []
    ) {
        self.id = id
        self.rawText = rawText
        self.parsedJson = parsedJson
        self.status = status
        self.createdAt = createdAt
        self.exercises = exercises
    }
}

public enum ScribbleStatus: String, Codable, Sendable {
    case raw
    case inProgress
    case parsed
    case completed
    case failed
}
